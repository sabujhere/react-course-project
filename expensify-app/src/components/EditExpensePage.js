import React from 'react';

  
const EditExpensePage = (props) => {
  console.log(props);
  return (
    <div>
      Editing the expenses for the ID of {props.match.params.id}
    </div>
  );
}

  export default EditExpensePage;