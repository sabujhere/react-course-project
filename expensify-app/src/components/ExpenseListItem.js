import React from 'react';
import {removeExpense} from '../actions/expenses';
import { connect } from 'react-redux';

const ExpenseListItem = ({dispatch, id,description, amount, createdAt}) => (
  <div>
    <h2>{description}</h2>
    <p>{amount} - {createdAt}</p>
    <button onClick={(e)=>{
      dispatch(removeExpense({id}))
    }}>Remove</button>
  </div>
  );

  
  export default connect()(ExpenseListItem);