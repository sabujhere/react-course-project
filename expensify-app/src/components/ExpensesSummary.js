import React from 'react';
import { connect } from 'react-redux';
import {Link} from 'react-router-dom';
import numeral from 'numeral';
import selectExpenseTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => (
    <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">Viewing <span>{expensesCount}</span> expense{expensesCount === 0 ? '':'s'} totalling <span>{numeral(expensesTotal/100).format('$0,0.00')}</span></h1>
        <div className="page-header__actions">
        <Link className="button" to="/create">Add Expense</Link>
        </div>
        </div>
    </div>
);

//export default ExpensesSummary;
const mapStateToProps = (state) =>{
    const filteredExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: filteredExpenses.length,
        expensesTotal: selectExpenseTotal(filteredExpenses)
    }
}

 export default connect(mapStateToProps)(ExpensesSummary);