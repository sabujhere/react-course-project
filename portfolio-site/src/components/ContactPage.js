import React from 'react';

  const ContactPage = () => (
    <div>
      This is from my contact page
    </div>
  );

  export default ContactPage;