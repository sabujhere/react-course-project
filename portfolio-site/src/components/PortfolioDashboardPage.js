import React from 'react';

const PortfolioDashboardPage = () => (
    <div>
      <h3>Welcome</h3>
      <p>This is my site. Take a look around.</p>
    </div>
  );

  export default PortfolioDashboardPage;