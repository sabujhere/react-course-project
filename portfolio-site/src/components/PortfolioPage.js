import React from 'react';
import { Link } from 'react-router-dom';
const PortfolioPage = () => (
    <div>
      <h3>My Work</h3>
      <p>Check the following things I've done:</p>
      <p><Link to="/Project/1">Project 1</Link></p>
      <p><Link to="/Project/2">Project 2</Link></p>
    </div>
  );

  export default PortfolioPage;