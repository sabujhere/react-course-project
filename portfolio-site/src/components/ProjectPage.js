import React from 'react';


const ProjectPage = (props) => (
    <div>
      <h3>This is my project page with ProjectID:{props.match.params.id}</h3>
    </div>
  );

  export default ProjectPage;