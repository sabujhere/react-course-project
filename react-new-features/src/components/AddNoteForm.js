import React, {useState} from 'react';

const AddNoteForm = ({dispatchAddNote}) =>{
    const [title, setTitle] = useState('')
    const [body, setBody] = useState('')

    const addNote = (e) => {
        e.preventDefault()
        dispatchAddNote({
            type: 'ADD_NOTE',
            title,
            body
        })
        setTitle('')
        setBody('')
    }


    return (
      <div>
      <p>Add note</p>
      <form onSubmit={addNote}>
          <input value={title} onChange={(e) => setTitle(e.target.value)} />
          <textarea value={body} onChange={(e) => setBody(e.target.value)}></textarea>
          <button>add note</button>
      </form>
      </div>
    )
}

export {AddNoteForm as default}

