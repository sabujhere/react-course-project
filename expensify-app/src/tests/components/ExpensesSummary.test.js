import React from 'react';
import { shallow } from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should connectly render expenseSummay with 1 expense', () => {    
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={10025}/>);
    expect(wrapper).toMatchSnapshot();
  });
  

  test('should connectly render expenseSummay with 2 expenses', () => {   
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={8000000025}/>);
    expect(wrapper).toMatchSnapshot();
  });