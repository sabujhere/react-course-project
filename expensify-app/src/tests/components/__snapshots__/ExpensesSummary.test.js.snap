// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should connectly render expenseSummay with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       expense
      s
       totalling 
      <span>
        $100.25
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should connectly render expenseSummay with 2 expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       expense
      s
       totalling 
      <span>
        $80,000,000.25
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
